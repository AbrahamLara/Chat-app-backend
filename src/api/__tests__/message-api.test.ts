import { SuperAgentTest } from 'supertest';
import { Server } from 'http';
import { MOCK_JWT, mockServerAndAgent } from '../../test_utils/mock-utils';
import { mockTokenUtils } from '../../test_utils/file_mocks/mock-token-utils';

let server: Server;
let agent: SuperAgentTest;

const MOCK_CHAT_ID = 'CHAT_ID';

mockTokenUtils();

describe('Message endpoint', () => {
  beforeEach(
    mockServerAndAgent((mockServer, testAgent) => {
      server = mockServer;
      agent = testAgent;
    })
  );

  afterEach(async cb => {
    server.close(cb);
  });

  it('returns an error message if creating message with blank message', async () => {
    const response = await agent
      .post(`/api/message/${MOCK_CHAT_ID}`)
      .send({ message: '' })
      .set('Content-Type', 'application/json')
      .set('Authorization', `Bearer ${MOCK_JWT}`);

    expect(response.ok).toEqual(false);
    expect(response.status).toEqual(400);
    expect(response.body).toMatchSnapshot();
  });
});
