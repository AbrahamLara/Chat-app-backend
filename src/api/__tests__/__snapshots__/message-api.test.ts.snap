// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message endpoint returns an error message if creating message with blank message 1`] = `
Object {
  "message": "Chat message must be a non-empty string.",
}
`;
